Quick-Start/Anatomy_of_extension.md

```diff
@@ -1,418 +1,418 @@
+     
+    // Return the result 
+    PG_RETURN_INT32(result); 
+} 
+ 
+``` 
+ 
+In this example: 
+-   The code file includes necessary header files (`postgres.h` and `fmgr.h`) that provide access to the PostgreSQL Extension API. 
+-   `PG_MODULE_MAGIC` is a macro that indicates the compatibility version of the PostgreSQL Extension API. It must be included in the code file. 
+-   `PG_FUNCTION_INFO_V1` is a macro that defines the information for an extension function. It is used to declare the function to the PostgreSQL server. 
+-   `my_extension_func` is the implementation of the extension's function. It takes two integer arguments, adds them together, and returns the result. 
+-   `PG_GETARG_INT32` is a macro that retrieves the function arguments of type `int32` (integer).   
+-   `PG_RETURN_INT32` is a macro that sets the return value of the function to an `int32` value. 
+    
+The code file can include additional functions, data types, operators, and other database entities as required by the extension's functionality. It provides the foundation for extending PostgreSQL and allows developers to define custom behaviour that can be executed within the database. 
+ 
+## Creating Code Files for PostgreSQL Extensions: 
+Code Files: The code files contain the actual implementation of the extension's functionality. These files are typically written in a programming language such as C, C++, Python, or any other supported language. The code files interact with the database through the SQL interface and provide additional features and capabilities. Create a quick start on how to go about writing Code Files (how it will help the developer)while creating an Extension in Postgres. 
+
+Sure! Here's a quick start guide on writing code files for a PostgreSQL extension: 
+
+1.  **Choose a Programming Language:** Decide on the programming language you want to use for implementing your PostgreSQL extension. Common choices include C, C++, and Python. Ensure that your chosen language is supported by PostgreSQL and has the necessary libraries and bindings for interacting with the database.   
+
+2.  **Set Up the Development Environment:** Set up your development environment with the required tools and dependencies. This typically involves installing the necessary compilers, development libraries, and PostgreSQL development packages. 
+
+3.  **Create the Code File:** Create a new code file with a descriptive name that reflects the purpose of your extension. For example, if your extension is called "my\_extension," you could create a file named "my\_extension.c" for a C-based extension or "my\_extension.py" for a Python-based extension. 
+
+4.  **Include Required Header Files:** In the code file, include the necessary header files that provide access to the PostgreSQL Extension API and any other libraries or modules you will be using. The specific header files may vary depending on the programming language and libraries you are using. 
```

Comment:
For non-C languages it's usually some sort of package import. But even that's not necessary for many languages. [PL/Perl](https://www.postgresql.org/docs/current/plperl.html), for example, just has various functions available to it. In fact, All of the `PL/*` languages don't go into a separate source file, but into the SQL script.